pkg/db/queries.sql

```diff
@@ -75,3 +75,56 @@ FROM
 GROUP BY
 	originator_node_id;
 
+-- name: GetAddressLogs :many
+SELECT
+	a.address,
+	encode(a.inbox_id, 'hex') AS inbox_id,
+	a.association_sequence_id
+FROM
+	address_log a
+	INNER JOIN (
+		SELECT
+			address,
+			MAX(association_sequence_id) AS max_association_sequence_id
+		FROM
+			address_log
+		WHERE
+			address = ANY (@addresses::TEXT[])
+			AND revocation_sequence_id IS NULL
+		GROUP BY
+			address) b ON a.address = b.address
+	AND a.association_sequence_id = b.max_association_sequence_id;
+
+-- name: InsertAddressLog :one
+INSERT INTO address_log(address, inbox_id, association_sequence_id, revocation_sequence_id)
+	VALUES (@address, decode(@inbox_id, 'hex'), @association_sequence_id, @revocation_sequence_id)
+RETURNING
+	*;
+
+-- name: RevokeAddressFromLog :exec
+UPDATE
+	address_log
+SET
+	revocation_sequence_id = @revocation_sequence_id
+WHERE (address, inbox_id, association_sequence_id) =(
+	SELECT
+		address,
+		inbox_id,
+		MAX(association_sequence_id)
+	FROM
+		address_log AS a
+	WHERE
+		a.address = @address
+		AND a.inbox_id = decode(@inbox_id, 'hex')
+	GROUP BY
+		address,
+		inbox_id);
+
+-- name: GetLatestSequenceId :one
+SELECT
+	COALESCE(max(originator_sequence_id), 0)::BIGINT AS originator_sequence_id
+FROM
+	gateway_envelopes
+WHERE
+	originator_node_id = @originator_node_id;
```

Comment:
We originally added this as a hack, but I guess we can keep it permanently? It does make the server more robust against edge cases

Reply:
Even once we have checks in place to start the blockchain indexing from the last seen block, there might be partially completed blocks from a past session. I'm OK with leaving for now.